ipfs/evolution.md

```diff
@@ -1 +1,143 @@
+
+## Decision
+
+- Implement QUIC as a new protocol to solve issues with TCP.
+
+
+## Status
+
+- Implemented
+
+## Consequences
+
+- Fewer local resources: allow us to dial faster and keep more connections open.
+- Faster connection establishment: significantly reduce the latency of DHT queries.
+- Behaves better on lossy networks: QUIC handles multiplexing internally, dropping a single packets affects only the related stream.
+
+
+# ADR2: Performance Resources - Release 0.4.18
+
+## Context
+
+- Actual the resource of utlization are a problem when it use the IPFS becouse the use a lot of CPU and memory.
+
+## Decision
+
+- Changed two of our most frequently used datastructures, CIDs and Multiaddrs, to reduce allocation load.
+- Store CIDs encode as strings, instead of decoded in structs.
+- Changed many of our multiaddr parsing/processing/formatting functions to allocate less.
+
+
+## Status
+
+- Implemented
+
+## Consequences
+
+- Reduce CPU utilization when heavily using the DHT.
+- Improve memory usage when inserting CID.
+
+
+
+# ADR3: New DHT - Release 0.5.0
+
+## Context
+
+- Actual DHT have problems trying to connect to peers that cannot be reached wasting query time.
+- The logic of DHT query doesn't properly terminate when it hits the end of the query, trying always to reconnect.
+- The routing table are poorly maintened
+  
+## Decision
+
+- Dual DHT
+  - All IPFS nodes will now run two DHTs: one for the public internet WAN, and one for their local network LAN.
+
+- Query Logic
+  - Publishing IPNS & provider records.
+  - Resolving IPNS addresses.
+- Routing Tables
+  - Addresing the poorly maintained routing tables by:
+
+  - Reducing the probability that the connection manager will kill connections to peers in the routing table.
+  - Keeping peers in the routing table even if we get disconnected from them.
+  - Prioritizing useful peers that respond to queries fast.
+## Status
+
+- Implemented
+
+## Consequences
+
+
+- Connections terminate when hits the end of the query.
+- Improve tables logic.
+- Brings stability to connections.
+- A lot of cost wasted rewriting the most of the module DHT.
+
+# ADR4: QUIC by default - Release 0.6.0
+
+## Context
+
+- QUIC transport is enabled by default for both inbound and outbound connections. When connecting to new peers, libp2p will continue to dial all advertised addresses in parallel (tcp + quic), so if the QUIC connection fails, the connection should still succeed.
+
+## Decision
+
+- With the QUIC transport, the IPFS handshake takes two round trips. In the future, this will be improved for reduce it to one round trip.
+- The QUIC transport will open one UDP socket per listen address instead of one socket per connection.
+  
+
+## Status
+
+- Implemented
+
+## Consequences
+
+- Have to make two round trips for establish a connection and the handshake, this should be improved to one round trip. 
+
+# ADR5: Removing support for the SECIO security transport - Release 0.7.0
```

Comment:
This ADR5 is very well defined, the context is a context, the decision a decision 👍